### PR TITLE
feat: add a toggle to switch between "Near me" cards and "Trending Pandals" cards

### LINKED ISSUE #98: add a toggle to switch between "Near me" cards and "Trending Pandals" cards

### PR DESCRIPTION
**Pull request by Dyslex7c**
Closes #98

## Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/Pujo-Atlas-Kolkata/PujoAtlasKol-Web/blob/dev/CONTRIBUTING.md)
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Added a toggle to switch between Near Me Pandals Cards and Trending Pandals Cards

---

## Screenshots

image

### CODE REVIEW

File: src/components/app/shared/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+import PandalCard from './PandalCard';
+import { TbLocationFilled } from 'react-icons/tb';
+import { IoMdTrendingUp } from 'react-icons/io';
+
+const Card = () => {
+  const [activeCard, setActiveCard] = useState('trending');
```
Comment (wriddhi) [inline src/components/app/shared/Card.tsx:7]: add a generic here so people get autocomplete in future

`const [activeCard, setActiveCard] = useState<"trending"|"nearme">("trending");`

Comment (Dyslex7c) [reply src/components/app/shared/Card.tsx:7]: okay okay

File: src/components/app/shared/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <>
+      <div className="flex flex-row gap-2">
+        <button
+          className={`p-2 px-3 font-work rounded-full bg-[${activeCard === 'trending' ? '#fff' : '#e6dfcf'}]`}
+          onClick={() => (activeCard !== 'nearme' ? setActiveCard('nearme') : null)}
+        >
+          Near Me
+        </button>
+        <button
+          className={`p-2 px-3 font-work rounded-full bg-[${activeCard === 'trending' ? '#e6dfcf' : '#fff'}]`}
```
Comment (AyanavaKarmakar) [inline src/components/app/shared/Card.tsx:18]:
```suggestion
          className={`p-2 px-3 font-sans rounded-full bg-[${activeCard === 'trending' ? '#e6dfcf' : '#fff'}]`}
```

File: src/components/app/shared/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { TbLocationFilled } from 'react-icons/tb';
+import { IoMdTrendingUp } from 'react-icons/io';
+
+const Card = () => {
+  const [activeCard, setActiveCard] = useState<'trending' | 'nearme'>('trending');
+  return (
+    <>
+      <div className="flex flex-row gap-2">
+        <button
+          className={`p-2 px-3 font-work rounded-full bg-[${activeCard === 'trending' ? '#fff' : '#e6dfcf'}]`}
```
Comment (AyanavaKarmakar) [inline src/components/app/shared/Card.tsx:12]:
```suggestion
          className={`p-2 px-3 font-sans rounded-full bg-[${activeCard === 'trending' ? '#fff' : '#e6dfcf'}]`}
```

Comment (AyanavaKarmakar) [inline src/components/app/shared/Card.tsx:12]: use `cn` for `className` conditionals. e.g - https://github.com/Pujo-Atlas-Kolkata/PujoAtlasKol-Web/blob/dev/src/components/app/shared/BottomNav.astro

File: src/components/app/shared/Card.tsx (hunk 2 above)
Comment (AyanavaKarmakar) [inline src/components/app/shared/Card.tsx:18]: use `cn` for `className` conditionals. e.g - https://github.com/Pujo-Atlas-Kolkata/PujoAtlasKol-Web/blob/dev/src/components/app/shared/BottomNav.astro

File: src/pages/app/index.astro [hunk 4]
```diff
@@ -14,6 +14,6 @@ import { cn } from '@/libs/utils';
     )}
   >
     <SearchBar />
-    <NearMeCard />
+    <Card client:load />
```
Comment (AyanavaKarmakar) [inline src/pages/app/index.astro:17]:
```suggestion
    <Card client:load client:only="react" />
```

**Review by AyanavaKarmakar — APPROVED** [review]

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]
^

**Review by AyanavaKarmakar — APPROVED** [review]

**Review by wriddhi — APPROVED** [review]

**Pull request merged**